src/vsphere_cpi/lib/cloud/vsphere/cloud.rb

```diff
@@ -506,15 +506,38 @@ def delete_disk(raw_director_disk_cid)
 
     # Replicating a stemcell allows the creation of linked clones which can share files with a snapshot.
     # For details see https://www.vmware.com/support/ws5/doc/ws_clone_overview.html.
-    def replicate_stemcell(cluster, to_datastore, stemcell_id)
+    def replicate_stemcell(cluster, to_datastore, stemcell_id, datastore_cluster=nil)
       original_stemcell_vm = self.client.find_vm_by_name(@datacenter.mob, stemcell_id)
       raise "Could not find VM for stemcell '#{stemcell_id}'" if original_stemcell_vm.nil?
 
+      #if datastore cluster, then find the recommended datastore and then proceed as usual
+      if datastore_cluster
+        name_of_replicated_stemcell = "#{stemcell_id} %2f #{SecureRandom.uuid}"
+        @logger.info("Replicating #{stemcell_id} (#{original_stemcell_vm}) to #{name_of_replicated_stemcell}")
+        recommendation = get_recommendation_for_stemcell(
+          original_stemcell_vm,
+          name_of_replicated_stemcell,
+          @datacenter.template_folder.mob,
+          cluster.resource_pool.mob,
+          datastore_cluster: datastore_cluster
+        )
+        @logger.info("Recommendation from Storage DRS for: #{name_of_replicated_stemcell}, Destination: #{recommendation.action.first.destination.name}")
+        #loop over all actions to pick one with reason as storagePlacement?
+        if recommendation.reason == 'storagePlacement' && recommendation.action.first.destination.class == VimSdk::Vim::Datastore
+          to_datastore = recommendation.action.first.destination
```

Comment:
could there be other recommendations (not first)?

Reply:
RecommendDatastores can return multiple recommendations, the best one will be first entry in the array.